### PR TITLE
Submission unit cycle

### PR DESCRIPTION
**Pull request by phlax**
clean up of UnitLifecycle to ensure that subs are created in the correct order and with correct data

### CODE REVIEW

File: pootle/apps/pootle_store/utils.py [hunk 1]
```diff
@@ -391,54 +391,64 @@ def save_subs(self, subs):
...
+            old_value=self.original.target or "",
             new_value=self.unit.target_f)
         _kwargs.update(kwargs)
         return self.create_submission(**_kwargs)
 
     def sub_state_update(self, **kwargs):
+        is_review = (
+            self.unit.change.reviewed_on
+            and (self.unit.change.reviewed_on
+                 >= self.unit.change.submitted_on))
```
Comment (unho) [inline pootle/apps/pootle_store/utils.py]: This looks like it can be a property of `unit.change`.

File: tests/pootle_store/utils.py [hunk 2]
```diff
@@ -171,14 +200,13 @@ def test_unit_lifecycle_update_target(store0, member):
...
-    unit.submitted_by = member
-    unit.save()
+    unit.save(submitted_by=member)
     sub_target_update = lifecycle.get(Unit)(unit).sub_target_update()
     assert isinstance(sub_target_update, Submission)
     assert sub_target_update.unit == unit
     assert sub_target_update.translation_project == store0.translation_project
     assert sub_target_update.revision == unit.revision
-    assert sub_target_update.submitter == unit.submitted_by
+    assert sub_target_update.submitter == unit.change.submitted_by
```
Comment (unho) [inline tests/pootle_store/utils.py]: No `assert` to ensure it is `member`?

**Comment by unho** [comment]
Nits aside, lgtm.

File: pootle/apps/pootle_store/utils.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_store/utils.py]: no - unit change is a model - and i think that would be confusing

File: tests/pootle_store/utils.py (hunk 2 above)
Comment (phlax) [inline tests/pootle_store/utils.py]: i can add

File: pootle/apps/pootle_store/utils.py (hunk 1 above)
Comment (unho) [inline pootle/apps/pootle_store/utils.py]: Models have convenience properties, but I can live with this as is.

**Pull request merged**